### ISSUE TITLE
vim editor permanently clobbers notes when switching between notes

### ISSUE THREAD

**Issue by hkhanna** [standardnotes/forum#985]
Just bought 1 year of StandardNotes and stumbled on a troubling bug that makes me worried about data integrity. 

**Describe the bug**
In the web application, if >1 note has the vim editor activated, and you click the first note, then the second note, then edit that second note, it will destroy that second note and make the contents irretrievable even via session history.

**To Reproduce**
1. Activate vim editor in a note.
2. Activate vim editor in a second note.
3. Click the first note, then the second note.
4. You'll see the contents of the second note seem like they haven't been updated and it's still displaying the first note.
5. If you accidentally edit the second note now, it will clobber the contents of the second note with the contents of the first note. 

**Expected behavior**
Switching between notes should properly update the displayed contents.

**Screenshots**
Step 1.
![Screen Shot 2020-05-23 at 3 32 10 PM](https://user-images.githubusercontent.com/222784/82739076-bdeed080-9d0a-11ea-8303-a898ea277705.png)

Step 2.
![Screen Shot 2020-05-23 at 3 32 19 PM](https://user-images.githubusercontent.com/222784/82739084-ca732900-9d0a-11ea-8f8d-5a44267e1016.png)

Step 3.
![Screen Shot 2020-05-23 at 3 32 40 PM](https://user-images.githubusercontent.com/222784/82739090-cfd07380-9d0a-11ea-8964-5e67d08f4236.png)

**Web**
 - OS: Mac OS X 10.14.6
 - Browser: error is on both Chrome (81.0.4044.138) and Firefox (76.0.1)

**Comment by baptiste-grob** [comment]
Hey @hkhanna, thank you very much for the detailed report. I couldn't reproduce the issue you're describing, so if you don't mind I have a couple of followup questions:
- Do you happen to have any browser add-ons enabled, that would interfere with the loading/unloading of the editor?
- Could you show the contents of the console when you are reproducing the bug? On Firefox, you can toggle it by going to Tools -> Web Development -> Web Console.

**Comment by hkhanna** [comment]
Thanks for the quick response @baptiste-grob! Here's a screenshot of the console. 
<img width="1433" alt="Screen Shot 2020-05-24 at 9 54 18 AM" src="https://user-images.githubusercontent.com/222784/82755914-c7c51200-9da4-11ea-95d5-e3e67b740959.png">

I don't believe any of my extensions would interfere with the editor. I have an adblocker sometimes, but I turned that off and it still doesn't work.

**Comment by baptiste-grob** [comment]
Thanks for the screenshot!
Hm. The last three errors aren't showing up on my end, so there's definitely something strange going on. Can you reproduce the bug on https://standardnotes.org/demo? If yes then this could indicate there is something specific to your setup that I'm not catching. If it works fine in the demo, you could try uninstalling, then reinstalling the Vim Editor.
I would also recommend trying to reproduce the bug with other editors, as well as (if you can) the desktop app. Does something similar happen in these scenarios?

**Comment by hkhanna** [comment]
Tried it on the demo, and the vim editor worked perfectly, although note that I was still getting the `Content Security Policy: The page's settings blocked the loading of a resource...` error in the console. But it worked!

So I tried uninstalling the vim editor and then reinstalling it in my own setup, and now it works. 

I haven't installed other editors yet-- still just using the web. But it seems to be working now. So, if someone else stumbles on this issue and is reading this, just try uninstalling the vim editor, refreshing your browser, and reinstalling.  I will post in this issue if I have further trouble with this extension! Thank you!
